### PR TITLE
feat(category): add getByUri driver call

### PR DESCRIPTION
**Pull request by griest024**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/graycoreio/daffodil/blob/master/CONTRIBUTING.md#commit
- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

```
[ ] Bugfix
[x] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[ ] Documentation content changes
[ ] Other... Please describe:
```

## Does this PR introduce a breaking change?
```
[ ] Yes
[x] No
```

## Other information

### CODE REVIEW

File: libs/category/driver/magento/src/category.service.spec.ts [hunk 1]
```diff
@@ -250,6 +256,123 @@ describe('Driver | Magento | Category | CategoryService', () => {
...
+
+      const productsOp = controller.expectOne(MagentoGetProductsQuery);
+      productsOp.flushData(mockGetProductsResponse);
+
+      expect(productsOp.operation.variables.filter.category_id.eq).toEqual(String(mockMagentoCategory.id));
+
+      flush();
+    }));
+
+    describe('when filters are requested', () => {
```
Comment (lderrickable) [inline libs/category/driver/magento/src/category.service.spec.ts:330]: I think this should read "when the request applied a filter" or something like that. Filters are always requested/retrieved from the backend.

**Review by lderrickable — CHANGES_REQUESTED** [review]

**Comment by griest024** [comment]
Blocked by #1462

File: libs/category/driver/magento/src/category.service.spec.ts (hunk 1 above)
Comment (griest024) [reply libs/category/driver/magento/src/category.service.spec.ts]: changed

**Review by lderrickable — DISMISSED** [review]

File: libs/category/driver/magento/src/category.service.ts [hunk 2]
```diff
@@ -91,6 +89,38 @@ export class DaffMagentoCategoryService implements DaffCategoryServiceInterface
     );
   }
 
+  getByUri(categoryRequest: DaffCategoryUriRequest): Observable<DaffGetCategoryResponse> {
+    const truncatedUri = categoryRequest.uri.replace('.html', '');
```
Comment (damienwebdev) [inline libs/category/driver/magento/src/category.service.ts:93]: I don't think we should be truncating the URI.

**Review by damienwebdev — CHANGES_REQUESTED** [review]

File: libs/category/driver/magento/src/category.service.ts (hunk 2 above)
Comment (griest024) [reply libs/category/driver/magento/src/category.service.ts:93]: I have a PR out for appending the suffix to the URL, which I think it required? Or do you think that we should have the URL suffix in the daff urls at all?

Comment (lderrickable) [reply libs/category/driver/magento/src/category.service.ts:93]: Why would magento expect a url without the suffix, but provide the suffix in the category response? Would it be better for daffodil to only deal with whatever format the platform expects in the uri query? It seems like with this PR and the suffix PR the end result is just us putting the suffix onto the uri only to remove it just before the request is made. Do we need the suffix for some reason?

Comment (griest024) [reply libs/category/driver/magento/src/category.service.ts:93]: yeah I was under the impression that there was a compelling reason to need the suffix app-side. There is some consequence for routing behavior but I don't remember what it is. @damienwebdev do you know?

Comment (damienwebdev) [reply libs/category/driver/magento/src/category.service.ts:93]: In retrospect, it's more correct to say that this truncating is configurable by the app-dev. It's not a safe assumption that the suffix is `.html`.

Comment (griest024) [reply libs/category/driver/magento/src/category.service.ts]: changed truncation behavior to be dynamic based on config option

**Review by damienwebdev — APPROVED** [review]
lgtm

**Pull request merged**